pkg/registry/ggcr_image.go

```diff
@@ -0,0 +1,109 @@
+package registry
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/pkg/errors"
+)
+
+type GGCRImage struct {
+	image    v1.Image
+	repoName string
+}
+
+func NewGGCRImage(repoName string, keychain authn.Keychain) (*GGCRImage, error) {
```

Comment:
Do we want to call it `GetRemoteImage` or something similar, because we do not want to create an image in case one doesn't exist right? We only want to fetch an already existing one.